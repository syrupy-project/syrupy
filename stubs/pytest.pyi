from typing import (
    Any,
    Callable,
    TypeVar,
)

ReturnType = TypeVar("ReturnType")

def fixture(func: Callable[..., ReturnType]) -> Callable[..., ReturnType]: ...

class Function: ...
class Item: ...
