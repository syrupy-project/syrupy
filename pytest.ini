[pytest]
addopts = -p pytester -p syrupy
