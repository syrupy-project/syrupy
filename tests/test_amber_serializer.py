from collections import namedtuple

import pytest


def test_non_snapshots(snapshot):
    with pytest.raises(AssertionError):
        assert "Lorem ipsum." == "Muspi merol."


def test_empty_snapshot(snapshot):
    assert snapshot == None  # noqa: E711
    assert snapshot == ""


@pytest.mark.parametrize(
    "actual",
    [
        "",
        r"Raw string",
        r"Escaped \n",
        r"Backslash \u U",
        "ü•ûüêçüçØ",
        "singleline:",
        "- singleline",
        "multi-line\nline 2\nline 3",
        "multi-line\nline 2\n  line 3",
    ],
    ids=lambda x: "",
)
def test_string(snapshot, actual):
    assert snapshot == actual


def test_multiple_snapshots(snapshot):
    assert "First." == snapshot
    snapshot.assert_match("Second.")
    assert snapshot == "Third."


@pytest.mark.parametrize(
    "actual",
    [
        {"b": True, "c": "Some text.", "d": ["1", 2], "a": {"e": False}},
        {"b": True, "c": "Some ttext.", "d": ["1", 2], "a": {"e": False}},
    ],
)
def test_dict(snapshot, actual):
    assert actual == snapshot


def test_set(snapshot):
    assert snapshot == {"this", "is", "a", "set"}
    assert snapshot == {"contains", "frozen", frozenset({"1", "2"})}


ExampleTuple = namedtuple("ExampleTuple", ["a", "b", "c", "d"])


def test_tuple(snapshot):
    assert snapshot == ("this", "is", ("a", "tuple"))
    assert snapshot == ExampleTuple(a="this", b="is", c="a", d={"named", "tuple"})


def test_numbers(snapshot):
    assert snapshot == 3.5
    assert snapshot == 7
    assert snapshot == 2 / 6


def test_list(snapshot):
    assert snapshot == [1, 2, "string", {"key": "value"}]


list_cycle = [1, 2, 3]
list_cycle.append(list_cycle)

dict_cycle = {"a": 1, "b": 2, "c": 3}
dict_cycle.update(d=dict_cycle)


@pytest.mark.parametrize("cyclic", [list_cycle, dict_cycle])
def test_cycle(cyclic, snapshot):
    assert cyclic == snapshot


class TestClass:
    def test_name(self, snapshot):
        assert snapshot == "this is in a test class"
