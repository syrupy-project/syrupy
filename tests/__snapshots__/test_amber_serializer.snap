# name: TestClass.test_name
    this is in a test class
---
# name: test_dict[actual0]
    {'b': True, 'c': 'Some text.', 'd': ['1', 2], 'a': {'e': False}}
---
# name: test_dict[actual1]
    {'b': True, 'c': 'Some ttext.', 'd': ['1', 2], 'a': {'e': False}}
---
# name: test_empty_snapshot
    None
---
# name: test_empty_snapshot.1
    
---
# name: test_multiple_snapshots
    First.
---
# name: test_multiple_snapshots.1
    Second.
---
# name: test_multiple_snapshots.2
    Third.
---
# name: test_set
    {'is', 'this', 'a', 'set'}
---
# name: test_string[0]
    
---
# name: test_string[1]
    Raw string
---
# name: test_string[2]
    Escaped \n
---
# name: test_string[3]
    Backslash \u U
---
# name: test_string[4]
    🥞🐍🍯
---
# name: test_string[5]
    singleline:
---
# name: test_string[6]
    - singleline
---
# name: test_string[7]
    multi-line
    line 2
    line 3
---
# name: test_string[8]
    multi-line
    line 2
      line 3
---
# name: test_tuple
    ('this', 'is', ('a', 'tuple'))
---
# name: test_tuple.1
    ExampleTuple(a='this', b='is', c='a', d={'named', 'tuple'})
---
